modules/ocf_irc/manifests/ircd.pp

```diff
@@ -0,0 +1,32 @@
+class ocf_irc::ircd {
+  package { 'inspircd':; }
+
+  service { 'inspircd':
+    ensure  => 'running',
+    restart => 'service inspircd reload',
+    require => Package['inspircd'],
+  }
```

Comment:
(for context, I've seen this get out-of-control in some Puppet codebases where Puppet is practically used as a process supervisor; a server isn't considered "ready" after a reboot until Puppet has had a chance to run)